lib/crawler.js

```diff
@@ -1241,11 +1234,20 @@ Crawler.prototype.handleResponse = function(queueItem, response, timeCommenced)
     crawler.emit("fetchheaders", queueItem, response);
 
     // Ensure response length is reasonable...
-    responseLength =
-        responseLength > 0 ? responseLength : crawler.maxResourceSize;
-
+    responseLength = responseLength > 0 ? responseLength : crawler.maxResourceSize;
     queueItem.stateData.contentLength = responseLength;
 
+    function emitFetchComplete(responseBody, decompressedBuffer) {
+        responseBody = crawler.decodeResponses ? crawler.decodeBuffer(responseBody, stateData.contentType) : responseBody;
```

Comment:
On the flipside, there's definitely a reason for us to allow switching off response decoding — after all, iconv-lite might get it wrong (and has for me quite frequently) — reducing the page to gibberish.